javascript-source/core/patternDetector.js

```diff
@@ -37,13 +37,23 @@
      * @function hasLinks
      * @export $.patternDetector
      * @param {Object} event
-     * @param {boolean} [aggressive]
      * @returns {boolean}
      */
     function hasLinks(event) {
         return patterns.link.test(event.getMessage());
     }
 
+    /**
+     * @function getLinks
+     * @export $.patternDetector
+     * @param {Object} event
+     * @returns {string[]}
+     */
+    function getLinks(event) {
+        var matches = Array.from(event.getMessage().matchAll(patterns.link));
```

Comment:
You're using `matchAll` which is a JavaScript function against a Java String object, this will cause issues.